blueocean-dashboard/src/main/js/api/Branch.js

```diff
@@ -29,4 +29,18 @@ export default class Branch extends Pipeline {
         }
         return false;
     }
+
+    /**
+     * Check if two branch names are the same.
+     * Works around encoding mismatch bugs by encoding each branch name and comparing to the other.
+     * @param branchName1
+     * @param branchName2
+     * @returns {boolean}
+     * @private
+     */
+    _compareBranchNames(branchName1, branchName2) {
+        return branchName1 === branchName2 ||
+            encodeURIComponent(branchName1) === branchName2 ||
+            branchName1 === encodeURIComponent(branchName2);
+    }
 }
```

Comment:
I agree we should move them as soon as they are used in two different plugins or we are supremely confident that they will be widely used by developers. We need to do an extra "gulp bundle:watch + slink, npm publish" dance for any code in there, and it can be a bit of productivity hit.